pkg/scheduler/scheduler.go

```diff
@@ -160,39 +160,26 @@ type Config struct {
 
 // New create a new KubernetesScheduler
 func New(config Config) *KubernetesScheduler {
+	executorId := config.Executor.ExecutorId.GetValue()
+	executorUID, err := uid.Parse(executorId)
+	if err != nil {
```

Comment:
Fatal panics are idiomatic in "New" constructors. If we want to return an error we can wrap it in another static constructor that validates the config input and returns an error.